internal/webhooks/cluster.go

```diff
@@ -215,6 +215,21 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 		return allErrs
 	}
 
+	// MachineDeploymentTopology name should be defined.
+	if newCluster.Spec.Topology.Workers != nil {
```

Comment:
Would this check be better in check. MachineDeploymentTopologiesAreUniqueAndDefinedInClusterClass ? It's where we're doing the rest of the checks for the MDs (and it means we only have to parse the list once)

Reply:
It felt to me like it wouldn't fit into the name as the name is very specifically describing what it is validating.

WDYT about `MachineDeploymentTopologiesAreValidAndDefinedInClusterClass`

with that name, I aggree let's move it there.

Reply:
Yeah the name change is good with me - there's no need for it to be that explicit as long as the comment is clear.

Reply:
Fixed